wgsl/index.bs

```diff
@@ -728,36 +728,38 @@ type.
 </pre>
 * type must be `f32`, `i32` or `u32`
 
-### Read-only Storage Texture Types ### {#texture-ro}
+### Storage Texture Types ### {#texture-storage}
 
 A <dfn noexport>read-only texture</dfn> is a texture from which individual texels
-may be read, without the use of a sampler.
-See [[#texture-builtin-functions]].
+may be read, without the use of a sampler. A <dfn noexport>write-only texture</dfn>
+is a texture to which individual texels may be written. See [[#texture-builtin-functions]].
+The `read-only` and `write-only` is specified through a decorator when creating
+the texture variable.
 
 <pre class='def'>
-`texture_storage_ro_1d<image_storage_type>`
+`texture_storage_1d<image_storage_type>`
   %1 = OpTypeImage type(image_storage_type) 1D 0 0 0 2 image_storage_type
 
-`texture_storage_ro_1d_array<image_storage_type>`
+`texture_storage_1d_array<image_storage_type>`
   %1 = OpTypeImage type(image_storage_type) 1D 0 1 0 2 image_storage_type
 
-`texture_storage_ro_2d<image_storage_type>`
+`texture_storage_2d<image_storage_type>`
   %1 = OpTypeImage type(image_storage_type) 0 0 0 2 image_storage_type
 
-`texture_storage_ro_2d_array<image_storage_type>`
+`texture_storage_2d_array<image_storage_type>`
   %1 = OpTypeImage type(image_storage_type) 0 1 0 2 image_storage_type
 
-`texture_storage_ro_3d<image_storage_type>`
+`texture_storage_3d<image_storage_type>`
   %1 = OpTypeImage type(image_storage_type) 3D 0 0 0 2 image_storage_type
 </pre>
-* type(image_storage_type) is one of `f32`, `i32`, or `u32`, depending on the storage type.
+* type(image_storage_type) for a `read` texture is one of `f32`, `i32`, or `u32`, depending on the storage type.
 
-When mapping to SPIR-V, a read-only storage texture variable must also have a `NonWritable` decoration.
-For example:
+When mapping to SPIR-V, a read-only storage texture variable must also have a
```

Comment:
For a later editorial fix, we may want to separate the SPIR-V mappings out.  This is a bit non-trivial with the attributes and the image-storage-types.